### ISSUE TITLE
Vue Router hooks break after upgrading to 7.0.1

### ISSUE THREAD

**Issue by 06b** [vuejs/vue-class-component#316]
After upgrading to 7.0.1, it appears that Vue Router hooks such as beforeRouteEnter & beforeRouteLeave no longer get called.

The following appears to be one of the smallest reduced test cases I was able to recreate: https://github.com/06b/vue-class-component-router-bug

![vue-class-component-router-bug](https://user-images.githubusercontent.com/1302542/53768903-2a807400-3ea8-11e9-94f2-71d43f3dcc6a.gif)

Steps to reproduce:
1) Yarn Install
2) Yarn Serve
3) Upgrading the package "vue-class-component": "^6.3.2" to "vue-class-component": "^7.0.1"
4) Yarn Serve

**Comment by ktsn** [comment]
Because vue-property-decorator has another version of vue-class-component (^6.0.0) so that the hooks didn't registered correctly. vue-property-decorator need to be update its dependencies.
/cc @kaorun343

**Comment by kaorun343** [comment]
It's fixed by this PR.

https://github.com/kaorun343/vue-property-decorator/pull/178

**Comment by kaorun343** [comment]
@06b 

Please use the latest version `vue-property-decorator@8.0.0` .

**Comment by 06b** [comment]
Updated & Confirmed that this is now fixed.

**Comment by kimlisa** [comment]
@06b I am sorry to comment on a closed issue, but your project was most similar set up to mine (vue, ts, class-component), and I have no idea why your project works (I ran it), and mine doesn't. I tried everything I could find on internet. I put the hooks in a different file like the docs say to do, or put them in main.ts like you, or put the hooks directly in my app.vue like someone else suggested. I upgraded relevant packages to the latest. Nothing.

How did you create this project?

I used Vue CLI 3 to create this project and use npm.
